scripts/add_electricity.py

```diff
@@ -556,30 +576,65 @@ def add_nice_carrier_names(n, config):
 
     costs = load_costs(snakemake.input.tech_costs, snakemake.config['costs'], snakemake.config['electricity'], Nyears)
     ppl = load_powerplants(snakemake.input.powerplants)
+    
+    if "renewable_carriers" in snakemake.config['electricity']:    
+        renewable_carriers = set(snakemake.config['renewable'])
+    else: 
+        logger.warning("Missing key `renewable_carriers` under config entry `electricity`. "
+                       "In future versions, this will raise an error. "
+                       "Falling back to carriers listed under `renewable`.")
+        renewable_carriers = snakemake.config['renewable']
+
+    extendable_carriers = snakemake.config['electricity']['extendable_carriers']
+    if not (set(renewable_carriers) & set(extendable_carriers['Generator'])):
+        logger.warning("No renewables found in config entry `extendable_carriers`. "
+                       "In future versions, these have to be explicitely listed. "
+                       "Falling back to all renewables.")
+    
+    conventional_carriers = snakemake.config["electricity"]["conventional_carriers"]
+
 
     attach_load(n, snakemake.input.regions, snakemake.input.load, snakemake.input.nuts3_shapes,
                 snakemake.config['countries'], snakemake.config['load']['scaling_factor'])
 
     update_transmission_costs(n, costs, snakemake.config['lines']['length_factor'])
 
-    carriers = snakemake.config['electricity']['conventional_carriers']
-    attach_conventional_generators(n, costs, ppl, carriers)
+    attach_conventional_generators(n, costs, ppl, conventional_carriers, extendable_carriers)
```

Comment:
why not include the fancy capacity updates?

```suggestion
    attach_conventional_generators(n, costs, ppl, conventional_carriers, extendable_carriers, **snakemake.config["conventional"])
```